nist-interface/src/components/mapboxViewer/helper/index.jsx

```diff
@@ -1,29 +1,114 @@
+export const getLocationToZoom = (stations, stationCode) => {
+  if (stations.length<1) {
+      return null;
+  }
+  if (stationCode) {
+      let station = stations.find(station => station.id.includes(stationCode));
+      if (station) {
+          let { properties: {longitude, latitude} } = station;
+          let location = [longitude, latitude];
+          return location;
+      }
+  }
+  // TODO: ideally, if variance is less, then take mean center else take mode center
+  return getMeanCenterOfLocation(stations);
+}
 
-export const BASEMAP_STYLES = [
-    {
-      id: 'satellite',
-      label: 'Satellite',
-      mapboxId: 'cldu1cb8f00ds01p6gi583w1m',
-      thumbnailUrl: `https://api.mapbox.com/styles/v1/covid-nasa/cldu1cb8f00ds01p6gi583w1m/static/-9.14,38.7,10.5,0/480x320?access_token=${process.env.MAPBOX_TOKEN}`
-    },
-    {
-      id: 'dark',
-      label: 'Default dark',
-      mapboxId: 'cldu14gii006801mgq3dn1jpd',
-      thumbnailUrl: `https://api.mapbox.com/styles/v1/mapbox/dark-v10/static/-9.14,38.7,10.5,0/480x320?access_token=${process.env.MAPBOX_TOKEN}`
-    },
-    {
-      id: 'light',
-      label: 'Default light',
-      mapboxId: 'cldu0tceb000701qnrl7p9woh',
-      thumbnailUrl: `https://api.mapbox.com/styles/v1/mapbox/light-v10/static/-9.14,38.7,10.5,0/480x320?access_token=${process.env.MAPBOX_TOKEN}`
-    },
-    {
-      id: 'topo',
-      label: 'Topo',
-      mapboxId: 'cldu1yayu00au01qqrbdahb3m',
-      thumbnailUrl: `https://api.mapbox.com/styles/v1/covid-nasa/cldu1yayu00au01qqrbdahb3m/static/-9.14,38.7,10.5,0/480x320?access_token=${process.env.MAPBOX_TOKEN}`
-    }
-];
-  
-export const BASEMAP_ID_DEFAULT = 'satellite';
\ No newline at end of file
+export const getZoomLevel = (region, agency, stationCode, zoomLevelInQueryParam) => {
+  if (zoomLevelInQueryParam) {
+      // zoom-level in queryParam has highest precedence
+      return zoomLevelInQueryParam;
+  }
+  if (stationCode) {
+      // station-code present in queryParam has second highest precedence
+      return 6;
+  }
+  if (agency && region) {
+      return 5;
+  }
+  if (agency === "nist" && !region) {
+      // NIST dataset is for CONUS region, so zoom more to CONUS
+      return 4;
+  }
+  return 2;
+}
+
+export const getMeanCenterOfLocation = (stations) => {
+  // go through the stations and average the lat and lon to get the center
+  let latSum = 0;
+  let lonSum = 0;
+  stations.forEach((station) => {
+      let { properties: {longitude, latitude} } = station;
+      latSum += latitude;
+      lonSum += longitude;
+  });
+  let latCenter = latSum / stations.length;
+  let lonCenter = lonSum / stations.length;
+  return [lonCenter, latCenter];
+}
+
+export const getMarkerStyle = (index, markerClasses) => {
```

Comment:
isn't there a way to overlay an image with color rather than use multiple images for each color?

Reply:
About using a color overlay instead of using different images for different markers, we can do that using a CSS filter. 
I have some attraction towards the gradient. It seems like I like the shading effects on a gradient :sweat_smile:
Anyway, I am okay with the result generated by the filter overlay as well.

Reply:
Any thoughts on using a CSS filter on top of the marker image?

Reply:
Instead used a svg and changed its color.